.github/workflows/docker-publish.yml

```diff
@@ -1,82 +1,89 @@
 
 on:
   workflow_dispatch:
     inputs:
       light-client:
         description: 'aptos or ethereum'
         type: choice
+        required: true
         options:
           - aptos
           - ethereum
+      version:
+        description: "Image version"
+        required: true
+        type: string
 
+permissions:
+  id-token: write
+  contents: read
+  attestations: write
+  packages: write
+
+# Defines two custom environment variables for the workflow. These are used for the Container registry domain, and a name for the Docker image that this workflow builds.
 env:
-  REGISTRY: ghcr.io
+  REGISTRY: "ghcr.io"
+  NAMESPACE: "argumentcomputer"
+  IMAGE_NAME: "proof-server"
+# There is a single job in this workflow. It's configured to run on the latest available version of Ubuntu.
 jobs:
-  container:
+  build-and-push-image:
     runs-on: ubuntu-latest
-    # https://docs.github.com/en/actions/reference/authentication-in-a-workflow
+    # Sets the permissions granted to the `GITHUB_TOKEN` for the actions in this job.
     permissions:
-      id-token: write
-      packages: write
       contents: read
-    timeout-minutes: 120
+      packages: write
+      attestations: write
+      id-token: write
+      #
     steps:
       - name: Checkout repository
-        id: checkout
         uses: actions/checkout@v4
-
-      - name: Install Docker BuildX
-        uses: docker/setup-buildx-action@v2
-        id: buildx
-        with:
-          install: true
-
-      # Login against a Docker registry except on PR
-      # https://github.com/docker/login-action
-      - name: Log into registry ${{ env.REGISTRY }}
-        # Ensure this doesn't trigger on PR's
-        if: github.event_name != 'pull_request'
-        uses: docker/login-action@v2
+      # Uses the `docker/login-action` action to log in to the Container registry registry using the account and password that will publish the packages. Once published, the packages are scoped to the account defined here.
+      - name: Log in to the Container registry
+        uses: docker/login-action@65b78e6e13532edd9afa3aa52ac7964289d1a9c1
         with:
           registry: ${{ env.REGISTRY }}
           username: ${{ github.actor }}
-          password: ${{ secrets.REPO_TOKEN }}
-
-      # Extract metadata (tags, labels) for Docker
-      # https://github.com/docker/metadata-action
-      - name: Extract Docker metadata
+          password: ${{ secrets.GITHUB_TOKEN }}
+      # This step uses [docker/metadata-action](https://github.com/docker/metadata-action#about) to extract tags and labels that will be applied to the specified image. The `id` "meta" allows the output of this step to be referenced in a subsequent step. The `images` value provides the base name for the tags and labels.
+      - name: Extract metadata (tags, labels) for Docker
         id: meta
-        uses: docker/metadata-action@v4
+        uses: docker/metadata-action@9ec57ed1fcdbf14dcef7dfbe97b2010124a938b7
```

Comment:
i'm not sure why we need exact tags, rather than versions, here and above? 
I guess this is security from the context later in the image, but was not sure as I read these lines.
Could a comment provide that context to the reader?

Reply:
Sorry, exact tags come from the Github documentation on how to publish docker images. We should be able to revert to versions